spec/models/form_spec.rb

```diff
@@ -75,4 +75,31 @@
       end
     end
   end
+
+  describe "#email_confirmation_status" do
+    it "returns :not_started" do
+      expect(form.email_confirmation_status).to eq(:not_started)
+    end
+
+    it "with submission_email set and no FormSubmissionEmail, returns :email_set_without_confirmation" do
+      form.submission_email = "test@example.gov.uk"
+      expect(form.email_confirmation_status).to eq(:email_set_without_confirmation)
+    end
+
+    it "with FormSubmissionEmail code returns :sent" do
+      create :form_submission_email, form_id: form.id, temporary_submission_email: "test@example.gov.uk", confirmation_code: "123456"
+      expect(form.email_confirmation_status).to eq(:sent)
+    end
+
+    it "with FormSubmissionEmail with no code returns :confirmed" do
+      create :form_submission_email, form_id: form.id, temporary_submission_email: "test@example.gov.uk", confirmation_code: ""
+      expect(form.email_confirmation_status).to eq(:confirmed)
+    end
+
+    it "with FormSubmissionEmail with code and email matches forms returns :confirmed" do
+      form.submission_email = "test@example.gov.uk"
+      create :form_submission_email, form_id: form.id, temporary_submission_email: "test@example.gov.uk", confirmation_code: "123456"
+      expect(form.email_confirmation_status).to eq(:confirmed)
+    end
```

Comment:
Do all these `create...` need to be created, could we not use build instead?

Reply:
We agreed, that we would leave the implementation as is and work on refactoring the method thats calling `save` as its doing a lot for one method.